src/redux/pattern/pattern.sagas.js

```diff
@@ -0,0 +1,29 @@
+import { takeEvery, call, put } from 'redux-saga/effects';
+import { setPatterns } from './pattern.actions';
+
+import { getAllPatterns } from './pattern.operations';
+
+import { GET_PATTERNS } from './pattern.types';
+import { setError } from '../error/error.actions';
+import { push } from 'connected-react-router';
+import { setPatternLoading } from '../pattern/pattern.actions';
+
+export function* handlePatternsLoad() {
+  try {
+    yield put(setPatternLoading(true));
+    const patterns = yield call(getAllPatterns, 0, 1000);
+    yield put(setPatterns(patterns.items));
+    yield put(setPatternLoading(false));
+  } catch (error) {
+    yield call(handlePatternsErrors, error);
```

Comment:
You forgot to add `yield put(setPatternLoading(false));` in catch cell, for stop loading, when error was return.

Reply:
Thank you, done